explorer/data/prelude.carbon

```diff
@@ -264,6 +268,30 @@ impl String as GreaterEqWith(Self) {
   }
 }
 
+impl i32 as NotEqWith(Self) {
+  fn NotEq[me: Self](other: Self) -> bool {
+    var comp: Ordering = me.(CompareWith(i32).Compare)(other);
+    match (comp) {
+      case Ordering.Equivalent() => {
+        return false;
+      }
+    }
+    return true;
```

Comment:
This seems unnecessarily verbose, and also a little surprising: if we define `!=` in terms of another operation, `==` seems like a more natural choice than `CompareWith`.

```suggestion
  return not __intrinsic_int_eq(me, other);
```

Reply:
Changed!